opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,677 @@
+            process.append(pickle.load(f))
+            f.close()
+
+        # reading the files to get the probability, queries and passages
+        prob = []
+        query = []
+        passages = []
+        for j in range(0, num_file):
+            for dict_str in process[j]:
+                if "probability" in dict_str.keys():
+                    prob.append(dict_str["probability"])
+                if "passage" in dict_str.keys():
+                    passages.append(dict_str["passage"])
+                if "query" in dict_str.keys():
+                    query.append(dict_str["query"])
+
+        df = pd.DataFrame(
+            list(zip(prob, query, passages)), columns=["prob", "query", "passages"]
+        )
+
+        df = df.drop_duplicates(subset=["query"])
+        df["passages"] = df.apply(lambda x: self.qryrem(x), axis=1)
+
+        df = df.sample(frac=1)
+        return df
+
+    def load_sentence_transformer_example(
+        self, df, use_accelerate: bool = False
+    ) -> List[str]:
+        """
+        Description:
+        Create input data for training
+
+        Parameters:
+        df: pd.DataFrame
+            required for loading sentence transformer examples.
+        use_accelerate: bool = False,
+            Optional, use accelerate to fine tune model. Default as false to not use accelerator.
+        Return:
+            the list of train examples.
+        """
+
+        train_examples = []
+        print("Loading training examples... ")
+
+        if use_accelerate is False:
+            for i in tqdm(range(len(df)), total=len(df)):
+                train_examples.append(
+                    InputExample(
+                        texts=[
+                            df[i : i + 1]["passages"].values[0],
+                            df[i : i + 1]["query"].values[0],
+                        ]
+                    )
+                )
+        else:
+            #  ML: broken down the class into following
+            #  train_examples = tasb_dataset_vanilla(df)
+            queries = list(df["query"])
+            passages = list(df["passages"])
+            for i in tqdm(range(len(df)), total=len(df)):
+                train_examples.append([queries[i], passages[i]])
+        return train_examples
+
+    def train_model(
+        self,
+        train_examples: List[str],
+        model_url: str = None,
+        output_path: str = None,
+        output_model_name: str = None,
+        use_accelerate: bool = False,
+        learning_rate: float = 2e-5,
+    ):
+        """
+        Description:
+        Takes in training data and a sentence transformer url to train a custom semantic search model
+
+        Parameters:
+        train_examples:
+            required, input for the sentence transformer model training
+        model_url: str = None
+            optional,the url to download sentence transformer model, if None, default as 'sentence-transformers/msmarco-distilbert-base-tas-b'
```

Comment:
> download sentence transformer model

Will the model downloaded to some local folder? Suggest adding more details here.

From line 297 `model_url = "sentence-transformers/msmarco-distilbert-base-tas-b"` , seems this is not URL, it's huggingface model id or name ? And from line 310 `model = SentenceTransformer(model_url)` we only support sentence transformer model. I think we'd better explain this limitation in the doc.